src/common/docker-compose/index.ts

```diff
@@ -75,8 +75,12 @@ export const generate = async (dependency_manager: LocalDependencyManager): Prom
     }
 
     if (node instanceof ServiceNode) {
-      if (node.node_config.getCommand().length) compose.services[url_safe_ref].command = node.node_config.getCommand();
-      if (node.node_config.getEntrypoint().length) compose.services[url_safe_ref].entrypoint = node.node_config.getEntrypoint();
+      if (node.node_config.getCommand().length) { // docker-compose expects environment variables used in commands/entrypoints to be prefixed with $$, not $ in order to use variables local to the container
+        compose.services[url_safe_ref].command = node.node_config.getCommand().map(command_part => command_part.replace(/"\$/g, '"$$$'));
+      }
+      if (node.node_config.getEntrypoint().length) {
+        compose.services[url_safe_ref].entrypoint = node.node_config.getEntrypoint().map(entrypoint_part => entrypoint_part.replace(/"\$/g, '"$$$'));
```

Comment:
Did you confirm that this was a problem for entrypoints as well?

Reply:
Yeah it was also a problem with entrypoints